MSGViewer/src/main/java/net/sourceforge/MSGViewer/ViewerPanel.java

```diff
@@ -683,8 +688,18 @@ private void printRecipients(StringBuilder sb, RecipientType to, String s) {
     private static String asMailto(RecipientEntry recipient) {
         String name = recipient.getName();
         String email = recipient.getEmail();
-        if (isNotBlank(email)) {
-            return "<a href='mailto:" + email + "'>" + (isNotBlank(name) ? name + " &lt;" + email + "&gt;" : email) + "</a>";
+        String smtp = recipient.getSmtp();
+        String mailTo = "";
+
+        if (isNotBlank(email) && ViewerHelper.isValidEmail(email)) {
+            mailTo = email;
+        } else if (isNotBlank(smtp)) {
+            mailTo = smtp;
+        }
```

Comment:
That snippet could be extracted as a member method of RecipientEntry :
```
    public String mailTo() {
        if (isNotBlank(email) && ViewerHelper.isValidEmail(email)) {
            return email;
        } else if (isNotBlank(smtp)) {
            return smtp;
        }
        return "";
    }
```

Reply:
Ok, but all models in the app seem very lean (i.e. they don't much logic).
Should this be principle be inverted?

Reply:
It look fine to me that "mailTo" is a property of RecipientEntry. And i'm not specially pleased with ViewPanel having too much logic.
But you are right, this project doesnt have a clear design :)

Reply:
Ok. I'll change that.
I agree. Are you keen on moving the logic to "Service" classes?
What about using Dagger for dependency injection?

Reply:
Maybe it would be overkill for that project ?
I'll think about that, thanks for the suggestion.
And thanks for the pull request ;)